src/main/scala/outwatch/Pipe.scala

```diff
@@ -0,0 +1,67 @@
+package outwatch
+
+import cats.effect.IO
+import outwatch.Sink.{ObservableSink, SubjectSink}
+import rxscalajs.Observable
+
+
+trait PipeOps[-I, +O] { self : Pipe[I, O]  =>
+
+  def mapSink[I2](f: I2 => I): Pipe[I2, O] = Pipe(redirectMap(f), self)
```

Comment:
Sadly, with all these methods we cannot overload and return a `Handler` if `I` and `O` are the same type...

Reply:
@fdietze What do you think about having this:
```scala
  implicit class HandlerOps[T](self: Pipe[T, T]) {

    def imap[S](read: T => S)(write: S => T): Handler[S] = self.mapPipe(write)(read)

    def lens[S](seed: T)(read: T => S)(write: (T, S) => T): Handler[S] = {
      val redirected = self.redirect[S](_.withLatestFrom(self.startWith(seed)).map { case (a,b) => write(b,a) })
      Handler(redirected, self.map(read))
    }

   def transformHandler[S](read: Observable[T] => Observable[S])(write: Observable[S] => Observable[T]): Handler[S] =
      self.transformPipe(write)(read)
```

and leaving `type Handler[T] = Pipe[T, T]`. This way a `Handler` is a `Pipe`, `transformHandlerSink`, and `transformHandlerSource` will not be needed any more (will be the same as `transformSink` and `transformSource`).

Reply:
Good idea, I'll give it a try.

Reply:
Done. I think it's much cleaner now.